### ISSUE TITLE
Using Checkboxes with DetailView

### ISSUE THREAD

**Issue by SatanEnglish** [6pac/SlickGrid#347]
Need a way to use the expanding row and the checkboxes together.

Currently the row have issues when open that the rows in the checkbox plugin are some of the hidden rows in the expanding rows.

Waiting on the changes for the checkboxes to deal with disabled checkboxes then will need to find a way to deal with these.

@ghiscoding  input on how we can go about this would be great.

Thoughts so far an override functions so we can use the `id` of the data in the selected rows instead of it's index.

**Comment by SatanEnglish** [comment]
@ghiscoding I have both check box and expanding detailView on a row.

Now that I am trying to implement this I wonder if we need to put in a notify to `onSelectedRowsChanged`

When an expanding or colapsing as we are changing the rows position.

**Comment by ghiscoding** [comment]
Personally I think you're running after trouble to try to use both of these plugins at the same time. As for the `onSelectedRowsChanged`, that exist at the Grid level, while the RowDetail should be totally independent from all other plugin (you never know which user will mix the plugins and who won't mix, meaning using them together or not).

**Comment by SatanEnglish** [comment]
@ghiscoding
Yer I've made another copy of the checkbox plugin that uses a property on the row instead.
It has a method to look the rows up on some key so that I can check if it needs re-selecting (dataContext.id for example is the default).

**Comment by SatanEnglish** [comment]
@ghiscoding 
Whats you thoughts on updating the RowDetail plugin a little 

```diff
        /** Get the Row Detail padding (which are the rows dedicated to the detail panel) */
        var getPaddingItem = function (parent, offset) {
            var item = {};

-            for (var prop in _grid.getData()) {
-                item[prop] = null;
-            }
            item.id = parent.id + '.' + offset;

            // additional hidden padding metadata fields
-            //item[_keyPrefix + 'collapsed'] = true;
-            //item[_keyPrefix + 'isPadding'] = true;
            item[_keyPrefix + 'parent'] = parent;
+            item[_keyPrefix + 'isDetailRow'] = true;
-            //item[_keyPrefix + 'offset'] = offset;

            return item;
        }

```
Removing all the row properties that are never going to be used for an expanded FAKE row.
(Or did you leave these so formatters wouldn't die?)

Adding in another bool property so it's easy to determine that the row is a Detail Fake row `_isDetailRow`

Thoughts?

**Comment by ghiscoding** [comment]
I don't recall what these are for, it probably comes from the original code from that blog. Just remember there's 3 things to be careful of,
1. column position move,
2. column filter
3. column sort (however this one just collapse all rows since we never addressed that one). 

I think removing the `for` loop would be a problem, if you don't do that then you'll end up rendering the rows with data behind. To see what would happen live, just change the row detail background from white to transparent and you will see all the data of the opened row beeing copied on all rows of the row detail... kinda scary. I saw this myself when I was troubleshooting that feature, that is why I mention this.

For the other piece of code, you cannot remove `isPadding`, that is actually the flag that says this is a line of the row detail or what I think you can the Fake line. Why don't you use `isPadding`, I believe that is what you need

**Comment by SatanEnglish** [comment]
@ghiscoding
Removing the `for` actually makes the fake rows an empty object! (`var item = {};` 2 lines above)

Again this is the FAKE rows.
Everything in this method is putting stuff on the fake row.
`for` is putting an `null` for every property.
We only use the padding `collapsed` `isPadding` `offset` of the row that is REAL

Only thing used on the FAKE row is the `parent` which is what we use for determining it's an expanded row when filtering.... 
`_parent !== undefined` is likely what I should use as I've just checked and it's used in the filter setup.

However I do think we should remove all the data from the fake rows if possible currently it sets up all properties with null instead of just making an empty object with `id` and `parent`.

**Comment by ghiscoding** [comment]
I didn't re-read the entire post, but just to point out that the reason the 2 plugins don't work together is because they both try to use the 1st column to put their checkbox/icon. For example, Row Detail plugin uses [getColumnDefinition()](https://github.com/6pac/SlickGrid/blob/master/plugins/slick.rowdetailview.js#L600) to update first column with the toggle icon and `checkboxselectcolumn` plugin uses exactly the same [getColumnDefinition()](https://github.com/6pac/SlickGrid/blob/master/plugins/slick.checkboxselectcolumn.js#L253) technique.

**Comment by ghiscoding** [comment]
I'm now in the need to use both Row Detail & Checkbox Selector plugins at same time, I created a new Example that I will add as a PR soon. It all works, the only thing to note though is that you need to use the `syncGridSelection` method to keep correct row selection after a re-render or an invalidate row is triggered, without this it won't keep the selection or hides it like you said

```js
dataView.syncGridSelection(grid, true);
```

![Xjaa1lF4Xp](https://user-images.githubusercontent.com/643976/77219603-dd505e00-6b0d-11ea-990c-f9fad41f8a6f.gif)

**Comment by ghiscoding** [comment]
@SatanEnglish 
I actually went further and modified my new Example to include 3 plugins in the same demo. The 3 plugins are (RowDetail, RowMoveManager & CheckboxSelector). To achieve that I had to add 2 new options to the RowMoveManager not to conflict with the CheckboxSelector. If you're interested, you can see another animated in the new PR #474
